### PR TITLE
Propagate default milestoning dates for aggregate operations in projection

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary

- Related to https://github.com/finos/legend-studio/issues/1471
- handle propagation for aggregation functions. We will not remove the date since propagation is not supported

## How did you test this change?

https://user-images.githubusercontent.com/87973126/189183263-32341fc5-1109-4d29-affe-2fd66c027185.mp4

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

**Review by MauricioUyaguari — APPROVED** [review]

**Review by MauricioUyaguari — APPROVED** [review]

File: packages/legend-application-query/src/stores/QueryBuilderValueSpecificationBuilderHelper.ts [hunk 1]
```diff
@@ -175,6 +180,7 @@ export const buildPropertyExpressionChain = (
               nextExpression instanceof AbstractPropertyExpression
                 ? nextExpression
                 : undefined,
+              isDatePropagationNotSupported,
```
Comment (akphi) [inline packages/legend-application-query/src/stores/QueryBuilderValueSpecificationBuilderHelper.ts:183]: why do we need to pass this to `isDefaultDatePropagationSupported()` anymore? we can just stop here and do `!disableDatePropagation && isDefaultDatePropagationSupported(...)`

File: packages/legend-application-query/src/stores/QueryBuilderValueSpecificationBuilderHelper.ts [hunk 2]
```diff
@@ -138,6 +142,7 @@ const isDefaultDatePropagationSupported = (
 export const buildPropertyExpressionChain = (
   propertyExpression: AbstractPropertyExpression,
   queryBuilderState: QueryBuilderState,
+  isDatePropagationNotSupported?: boolean,
```
Comment (akphi) [inline packages/legend-application-query/src/stores/QueryBuilderValueSpecificationBuilderHelper.ts:145]: --> `disableDatePropagation`, let's also add documentation this flag. Every option-orientated param needs an explanation, else we could keep adding options like this on and on and at some point, we won't know what is each for anymore 😓

**Review by akphi — CHANGES_REQUESTED** [review]
Since we don't really file an issue for this, I don't know if this solve what issue, could we add some sort of a test for this? Thanks!

**Comment by MauricioUyaguari** [comment]
Agree with An. Additionally we should add tests for this PR and do some more investigation on what functions aren't supported for date propagation. 
I've create issue: https://github.com/finos/legend-studio/issues/1471

**Comment by akphi** [comment]
@gayathrir11 we could think of a way to also industrialize these tests, basically, these would be similar to lambda roundtrip test, but this is about rewriting query a bit right? I might be wrong, but let's figure out a way so we can systematically test these. Milestoning is complex and solidifying them is the only way to get things right eventually. Even #1471 should be carried out in this fashion I think, every new case we find, even if we don't decide to support that function, we need a test to back that decision

**Pull request merged**